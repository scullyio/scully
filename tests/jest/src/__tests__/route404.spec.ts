import { readPage, replaceIndexNG } from '../test-config.helper';
// import { title404 } from '../../../../libs/scully/src/lib/utils/serverstuff/title404';
import { expect } from '@jest/globals';

// copied in here, because the import blows up in GA.
const title404 = '404 - URL not provided in the app Scully is serving';

describe('baseroute 404 handling working', () => {
  const index: string = readPage('/test/fake1');
  const cleanIndex = replaceIndexNG(index);

  it('should match the snapshot', () => {
    expect(cleanIndex).toMatchSnapshot();
  });

  it(`Should have the text "${title404}"`, () => {
    expect(cleanIndex.includes(title404)).toBe(true);
  });
});
