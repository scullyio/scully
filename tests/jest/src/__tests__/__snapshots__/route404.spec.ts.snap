// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`baseroute 404 handling working should match the snapshot 1`] = `
<html>
  <head>
    <meta name="generator">
    <h1>
      404 - URL not provided in the app Scully is serving
    </h1>
    <p>
      The url "/test/fake1" is not provided in the scully.routes.json, so it can't be generated
    </p>
    <p>
      If you have routes that are not known before  you might want to serve index.html instead of this response.
      <br>
      That can be done by adding the '--404=index' option to the Scully command
    </p>
  </head>
</html>
`;
